Tasks/AzureFileCopy/Utility.ps1

```diff
@@ -94,34 +94,40 @@ function Get-AzureVersionComparison($azureVersion, $compareVersion)
 
 function Get-AzureUtility
 {
+    param([string][Parameter(Mandatory=$true)]$connectedServiceName)
+
     $currentVersion =  Get-AzureCmdletsVersion
     Write-Verbose "Installed Azure PowerShell version: $currentVersion"
 
     $AzureVersion099 = New-Object System.Version(0, 9, 9)
     $AzureVersion103 = New-Object System.Version(1, 0, 3)
+    $AzureVersion132 = New-Object System.Version(1, 3, 2)
 
     $azureUtilityVersion098 = "AzureUtilityLTE9.8.ps1"
     $azureUtilityVersion100 = "AzureUtilityGTE1.0.ps1"
     $azureUtilityVersion110 = "AzureUtilityGTE1.1.0.ps1"
+	$azureUtilityRest100 = "AzureUtilityRest.ps1"
 
     if(!(Get-AzureVersionComparison -AzureVersion $currentVersion -CompareVersion $AzureVersion099))
     {
-        $azureUtilityRequiredVersion = $azureUtilityVersion098
+		return $azureUtilityVersion098
     }
-    elseif(!(Get-AzureVersionComparison -AzureVersion $currentVersion -CompareVersion $AzureVersion103))
+	
+    if(!(Get-AzureVersionComparison -AzureVersion $currentVersion -CompareVersion $AzureVersion103))
     {
-        $azureUtilityRequiredVersion = $azureUtilityVersion100
+		return $azureUtilityVersion100
     }
-    else
+	
+	$endpoint = Get-TypeOfConnection $connectedServiceName
+	if(!(Get-AzureVersionComparison -AzureVersion $currentVersion -CompareVersion $AzureVersion132) -or ($endpoint.Auth.Scheme -eq "UserNamePassword"))
```

Comment:
This check won't honor the case where auth scheme is 'Certificate'. Can you reconfirm this scenario while performing the testing.